hibernate-core/src/test/java/org/hibernate/orm/test/embeddable/EmbeddableWithJavaTypeTest.java

```diff
@@ -48,10 +48,10 @@ public void injectSessionFactoryScope(SessionFactoryScope scope) {
 	@ValueSource(strings = {
 			"select z from EntityEmbedCustom z where embedCustom.value=:datum",
 			"select z from EntityEmbedCustom z where :datum=embedCustom.value",
-			"select z from EntityEmbedCustom z where embedCustom=:datum", // this query failed with the bug
-			"select z from EntityEmbedCustom z where :datum=embedCustom",
+			"select z from EntityEmbedCustom z where embedCustom.value=:datum", // this query failed with the bug
```

Comment:
It's a bit worrying that you're changing the query that was marked as "this query failed with the bug" to something that is already a contained in this array of strings. Seems like your change causes some change in behavior? The same concern applies to the other query changes here.